src/containers/Tag.js

```diff
@@ -73,12 +59,28 @@ class Tag extends Component {
   }
 
   render() {
-    const { lists, entities, params } = this.props
+    const { lists, entities, page, pathname, tagId } = this.props
     const postEntities = _.get(entities, reduxStateFields.postsInEntities, {})
-    const tagId = _.get(params, 'tagId')
     const error = _.get(lists, [ tagId, 'error' ], null)
+
+    // total items will be in that tagId
     const total = _.get(lists, [ tagId, 'total' ], 0)
-    const posts = utils.denormalizePosts(_.get(lists, [ tagId, 'items' ], []), postEntities)
+
+    // pages will be like
+    // {
+    //   1: [0, 9],
+    //   3: [10, 19],
+    // }
+    //
+    // which means the items of page 1 are in items[0] to items[9],
+    // the items of page 3 are in items[10] to item [19]
+    const pages = _.get(lists, [ tagId, 'pages' ], {})
+    const startPos = _.get(pages, [ page, 0 ], 0)
+    const endPos = _.get(pages, [ page, 1 ], 0)
+    // denormalize the items of current page
+    const posts = utils.denormalizePosts(_.get(lists, [ tagId, 'items' ], []).slice(startPos, endPos + 1), postEntities)
+
+    const totalPages = total / MAXRESULT
```

Comment:
`const totalPages = Math.ceil(total / MAXRESULT)` will be safer since `Pagination` might only be able to handle integer.